src/strings/nfa-noodlification.cc

```diff
@@ -235,14 +235,13 @@ std::vector<seg_nfa::NoodleWithEpsilonsCounter> seg_nfa::noodlify_mult_eps(const
         SegItem item = lifo.front();
         lifo.pop_front();
 
-        if(item.seg_id + 1 == segments.size()) {
+        if (item.seg_id + 1 == segments.size()) {
             // check if the noodle is already there
-            if(!std::any_of(noodles.begin(), noodles.end(),
-                [&](NoodleWithEpsilonsCounter &s) {
-                    return s == item.noodle;
-            } )) {
-                noodles.push_back(item.noodle);
-            }
+            if (!std::ranges::any_of(
+                    noodles
+                   ,
+                    [&](const NoodleWithEpsilonsCounter& s) { return s == item.noodle; }
+                    )) { noodles.push_back(item.noodle); }
```

Comment:
It is definitely an improvement. But the formatting still confuses me a bit. I think we can put the if condition on one line (the length would still be reasonable).

```suggestion
            if (!std::ranges::any_of(noodles, [&](const NoodleWithEpsilonsCounter& s) { return s == item.noodle; })) {
                 noodles.push_back(item.noodle); 
            }
```